test/unit/StreamrClient.test.js

```diff
@@ -806,6 +806,7 @@ describe('StreamrClient', () => {
         })
 
         it('rejects the promise if autoConnect is false and the client is not connected', (done) => {
+            client.options.auth.username = 'username'
```

Comment:
The new publish sequence now tries to create message before checking for valid connection (i.e. valid credentials), so this test would fail on invalid auth config before it even gets a chance to fail on the connection status.

Reply:
Just a note, that connecting (the websocket) does not trigger authentication sequence (`getSessionToken` does). Establishing a connection with invalid/no credentials is/should be possible.